packages/auth/src/next/helpers/user.ts

```diff
@@ -84,14 +69,12 @@ export async function getUser<
     return null;
   }
 
-  if (!ctx.callbacks?.onUser) {
-    return authenticatedUser;
-  }
-
-  const data = await ctx.callbacks.onUser(authenticatedUser, req);
-  if (!data) {
-    return authenticatedUser;
+  if (req && ctx.callbacks?.onUser) {
+    const data = await ctx.callbacks.onUser(authenticatedUser, req);
```

Comment:
@adam-maj why do we need to pass the request object to the callback functions?

Reply:
Users should be able to access the request object on the callback functions in case they need to run their own business logic with anything else on headers/body/etc.

Reply:
I see. So no need to pass it in App Router because users always have access to headers, cookies via next/headers package.